avaje-config/src/main/java/io/avaje/config/Configuration.java

```diff
@@ -240,6 +241,8 @@ public interface Configuration {
    */
   <T extends Enum<T>> T getEnum(Class<T> type, String key, T defaultValue);
 
+  <T> T getAs(String key, Function<CoreEntry, T> mappingFunction);
```

Comment:
CoreEntry is not public, so to me this does not make sense.

Reply:
tbh I was just copying how helidon does it, shall I make it use a regular string?